python-avd/tests/pyavd/j2filters/test_secure_hash.py

```diff
@@ -0,0 +1,39 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+
+from __future__ import annotations
+
+import re
+
+import pytest
+
+from pyavd.j2filters import secure_hash
+
+sha512_regex = r"^\$6\$[A-Za-z0-9\.\\\/]{16}\$[A-Za-z0-9\.\\\/]{86}"
+
+
+INVALID_PASSWORDS = [
+    pytest.param(True, TypeError, "Password MUST be of type 'str' but is of type <class 'bool'>"),
+    pytest.param([], TypeError, "Password MUST be of type 'str' but is of type <class 'list'>"),
+    pytest.param({}, TypeError, "Password MUST be of type 'str' but is of type <class 'dict'>"),
+]
+
+VALID_PASSWORDS = [
+    pytest.param("arista", sha512_regex),
+    pytest.param("rAnDoM123456789", sha512_regex),
+    pytest.param("ar!st@Us95r", sha512_regex),
+    pytest.param("superSecure999", sha512_regex),
+]
+
+
+class TestSecureHashFilter:
+    @pytest.mark.parametrize(("user_password", "expected_raise", "expected_raise_message"), INVALID_PASSWORDS)
+    def test_secure_hash_invalid(self, user_password: str, expected_raise: Exception, expected_raise_message: str) -> None:
+        with pytest.raises(expected_raise, match=expected_raise_message):
+            secure_hash(user_password)
+
+    @pytest.mark.parametrize(("user_password", "regex_of_hashed_password"), VALID_PASSWORDS)
+    def test_secure_hash_valid(self, user_password: str, regex_of_hashed_password: str) -> None:
+        hashed_password = secure_hash(user_password)
+        assert re.fullmatch(regex_of_hashed_password, hashed_password)
```

Comment:
Are these password hashes not identical between runs? I would rather that we put the hash values in the expected test data instead of the same regex on all of them.

If they are not consistent we need to reconsider this completely, since it could lead to constant config changes for every AVD run and never have "compliant" configs.

Reply:
Ack, the way the filter is now there will be config changes when AVD is re-run so I will change this. I'm working on adding a salt argument to the filter after an above discussion with @gmuloc, this will allow a user to configure the salt value used when the hash is created and this will ensure the hashed password does not change when AVD is run again. 

Regarding this test, it is just to confirm the hashed password is of a valid format i.e. `$6$<salt-value>$<hashed-password>`. Once the salt arg is done, I'll create some tests using hardcoded salt passwords against the expected hashed password.

Reply:
added the support for the optional salt arg:

```
local_users:
  # Create sha512 password hash with random salt. Note: this will create a new hash each time it is run
  - name: admin
    sha512_password: "{{ 'password123' | arista.avd.secure_hash }}"

  # Create sha512 password hash with user defined salt value
  - name: cvpadmin
    sha512_password: "{{ 'securepassword' | arista.avd.secure_hash(salt='Yar49ahkzKddRVYS') }}"
```